azurerm/internal/services/apimanagement/api_management_api_schema_resource.go

```diff
@@ -140,10 +156,32 @@ func resourceApiManagementApiSchemaRead(d *pluginsdk.ResourceData, meta interfac
 	if properties := resp.SchemaContractProperties; properties != nil {
 		d.Set("content_type", properties.ContentType)
 		if documentProperties := properties.SchemaDocumentProperties; documentProperties != nil {
-			d.Set("value", documentProperties.Value)
+			/*
+				As per https://docs.microsoft.com/en-us/rest/api/apimanagement/2019-12-01/api-schema/get#schemacontract
+
+				- Swagger Schema use application/vnd.ms-azure-apim.swagger.definitions+json
+				- WSDL Schema use application/vnd.ms-azure-apim.xsd+xml
+				- OpenApi Schema use application/vnd.oai.openapi.components+json
+				- WADL Schema use application/vnd.ms-azure-apim.wadl.grammars+xml.
+
+				Definitions used for Swagger/OpenAPI schemas only, otherwise Value is used
+			*/
+			switch *properties.ContentType {
+			case "application/vnd.ms-azure-apim.swagger.definitions+json", "application/vnd.oai.openapi.components+json":
+				if documentProperties.Definitions != nil {
+					value, err := json.Marshal(documentProperties.Definitions)
+					if err != nil {
+						return fmt.Errorf("[FATAL] Unable to serialize schema to json. Error: %+v. Schema struct: %+v", err, documentProperties.Definitions)
+					}
+					d.Set("value", string(value))
+				}
+			case "application/vnd.ms-azure-apim.xsd+xml", "application/vnd.ms-azure-apim.wadl.grammars+xml":
+				d.Set("value", documentProperties.Value)
+			default:
+				return fmt.Errorf("[FATAL] Unknown content type %q for schema %q (API Management Service %q / API %q / Resource Group %q)", *properties.ContentType, schemaID, serviceName, apiName, resourceGroup)
```

Comment:
instead of defaulting to an error can we just default to the odl behaviour?
```suggestion
			default:
				d.Set("value", documentProperties.Value)
```

Reply:
@katbyte Instead of having the same logic as you propose (having the advantage of being concise), I propose maybe a bit more verbose approach, but providing logging in case new content types will appear. Regardless if it will work or not, I hope that the log entry will help identify place requiring change.

Reply:
oh good call! i like it 🙂